controllers/metal3machinetemplate_controller.go

```diff
@@ -113,6 +119,32 @@ func (r *Metal3MachineTemplateReconciler) reconcileNormal(ctx context.Context,
 func (r *Metal3MachineTemplateReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&capm3.Metal3MachineTemplate{}).
+		Watches(
+			&source.Kind{Type: &capm3.Metal3Machine{}},
+			handler.EnqueueRequestsFromMapFunc(r.Metal3MachinesToMetal3MachineTemplate),
+		).
 		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
 		Complete(r)
 }
+
+func (m *Metal3MachineTemplateReconciler) Metal3MachinesToMetal3MachineTemplate(o client.Object) []ctrl.Request {
+	result := []ctrl.Request{}
+	// Fetch the Metal3MachineTemplate instance.
+	metal3MachineTemplate := &capm3.Metal3MachineTemplate{}
```

Comment:
I see you are retrieving the related/referenced template, but I think the part that make the template the source of truth is missing.

Reply:
That part is already in the main: https://github.com/metal3-io/cluster-api-provider-metal3/blob/13aa6e002eb43c938559a66a7e17328aa3390198/baremetal/metal3machinetemplate_manager.go#L95.
This patch intends to add a watch only.

Reply:
Got it, thanks.